pkgs/record_use/lib/src/record_use.dart

```diff
@@ -0,0 +1,128 @@
+import 'data_classes/reference.dart';
+import 'data_classes/usage.dart';
+import 'data_classes/usage_record.dart';
+
+extension type RecordedUsages._(UsageRecord _usages) {
+  RecordedUsages.fromJson(Map<String, dynamic> json)
+      : this._(UsageRecord.fromJson(json));
+
+  /// Show the metadata for this recording of usages.
+  Metadata get metadata => _usages.metadata;
+
+  /// Finds all const arguments for calls to the [method].
+  ///
+  /// The definition must be annotated with `@RecordUse()`. If there are no
+  /// calls to the definition, either because it was treeshaken, because it was
+  /// not annotated, or because it does not exist, returns `null`.
+  ///
+  /// Returns an empty iterable if the arguments were not collected.
+  ///
+  /// Example:
+  /// ```dart
+  /// import 'package:meta/meta.dart' show ResourceIdentifier;
+  /// void main() {
+  ///   print(SomeClass.someStaticMethod(42));
+  /// }
+  ///
+  /// class SomeClass {
+  ///   @ResourceIdentifier('id')
+  ///   static someStaticMethod(int i) {
+  ///     return i + 1;
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// Would mean that
+  /// ```
+  /// argumentsTo(Identifier(
+  ///           uri: 'path/to/file.dart',
+  ///           parent: 'SomeClass',
+  ///           name: 'someStaticMethod'),
+  ///       ).first ==
+  ///       Arguments(
+  ///         constArguments: ConstArguments(positional: {1: 42}),
+  ///       );
+  /// ```
+  Iterable<Arguments>? argumentsTo(Identifier method) => _callTo(method)
+      ?.references
+      .map((reference) => reference.arguments)
+      .whereType();
+
+  /// Finds all fields of a const instance of the class at [classIdentifier].
+  ///
+  /// The definition must be annotated with `@RecordUse()`. If there are
+  /// no instances of the definition, either because it was treeshaken, because
+  /// it was not annotated, or because it does not exist, returns `null`.
+  ///
+  /// The types of fields supported are defined at
+  ///
+  /// Example:
+  /// ```dart
+  /// void main() {
+  ///   print(SomeClass.someStaticMethod(42));
+  /// }
+  ///
+  /// class SomeClass {
+  ///   @AnnotationClass('freddie')
+  ///   static someStaticMethod(int i) {
+  ///     return i + 1;
+  ///   }
+  /// }
+  ///
+  /// @ResourceIdentifier()
+  /// class AnnotationClass {
+  ///   final String s;
+  ///   const AnnotationClass(this.s);
+  /// }
+  /// ```
+  ///
+  /// Would mean that
+  /// ```
+  /// instancesOf(Identifier(
+  ///           uri: 'path/to/file.dart',
+  ///           name: 'AnnotationClass'),
+  ///       ).first ==
+  ///       [
+  ///         Field(name: "s", className: "AnnotationClass", value: "freddie")
+  ///       ];
+  /// ```
+  ///
+  /// What kinds of fields can be recorded depends on the implementation of
+  /// https://dart-review.googlesource.com/c/sdk/+/369620/13/pkg/vm/lib/transformations/record_use/record_instance.dart
+  Iterable<List<Field>>? instancesOf(Identifier classIdentifier) =>
```

Comment:
There should be a class representing a constant instance e.g. `InstanceConstant` that has values for all fields.

Reply:
So basically `typedef InstanceConstant = List<Field>`?

Reply:
More like
```dart
class InstanceConstant {
  final Map<Identifier, Constant> fieldValues;
}
```
which could look like
```
{
  Identifier('package:foo/bar.dart', 'BaseClass', 'fieldA'): IntConstant(...),
  Identifier('package:foo/bar.dart', 'SubClass', 'fieldB'): StringConstantConstant(...),
}
```